src/components/spatialdisplay/SpatialDisplayComponent.vue

```diff
@@ -123,16 +143,42 @@ const colorScaleRangeString = computed(() => {
 })
 const legendLayerName = ref(props.layerName)
 const settings = useUserSettingsStore()
+
 const baseUrl = configManager.get('VITE_FEWS_WEBSERVICES_URL')
-const legendGraphic = useWmsLegend(
-  baseUrl,
-  legendLayerName,
-  () => settings.useDisplayUnits,
-  colorScaleRangeString,
-)
+const handleStyleClick = (style: Style) => {
+  selectedStyle.value = style
+}
+
+const updateColorScaleRange = (range: { min: number; max: number }) => {
+  colorScaleRange.value = range
+}
+
+const usedStyle = computed(() => {
+  if (selectedStyle.value !== undefined) {
+    return selectedStyle.value
+  } else if (props.layerCapabilities?.styles !== undefined) {
+    return props.layerCapabilities.styles[0]
+  } else {
+    return null
+  }
+})
+
+const legendGraphic = computed(() => {
```

Comment:
This does not need to be computed, the `loadLegend()` in `useWms` already gives back a `ref` legendGraphic which gets updated in the watchEffect below when `style`'s value changes.

```ts
export function useWmsLegend(
  baseUrl: string,
  layerName: MaybeRefOrGetter<string>,
  useDisplayUnits: MaybeRefOrGetter<boolean>,
  colorScaleRange?: MaybeRefOrGetter<string | undefined>,
  style?: MaybeRefOrGetter<Style | null>,
): Ref<GetLegendGraphicResponse | undefined> {
  ...
  const legendGraphic = ref<GetLegendGraphicResponse>()

  async function loadLegend(): Promise<void> {
    ...
    const _style = toValue(style)
    if (_layers === '') {
      legendGraphic.value = undefined
    } else {
      try {
        legendGraphic.value = await wmsProvider.getLegendGraphic({
           ...
          style: _style?.name,
        })
      } catch (error) {
        console.error(error)
      }
    }
  }

  watchEffect(() => {
    loadLegend()
  })
```